src/Lib/Element.php

```diff
@@ -11,18 +11,40 @@
  */
 abstract class Element implements \JsonSerializable
 {
-    /** @var array */
+    /** @var  Container $container */
+    protected $container;
+
+    /** @var array $options */
     protected $options = [];
 
-    /** @var array */
+    /** @var array $elements */
     protected $elements = [];
 
-    /** @var array */
+    /** @var array $externalOptions */
     protected $externalOptions = [];
 
-    /** @var array */
+    /** @var array $externalElements */
     protected $externalElements = [];
 
+    /**
+     * Set the dependency injection container / service locator
+     * @param Container $container
+     */
+    public function __construct(Container $container)
```

Comment:
Should this not be an instance of ```ContainerInterface``` and not your ```Container``` now? Otherwise what was the point :wink: